src/components/needle.js

```diff
@@ -1,12 +1,13 @@
-import React from 'react';
-import d3Shape from "d3-shape";
+import React from "react";
 
 export default class Needle extends React.Component {
+
   render() {
+    const {rotation, height} = this.props;
     return (
-      <g transform={`rotate(50)`}>
+      <g transform={`rotate(${rotation})`}>
         <path
-          d="M 0 5 C -1,5 -4,3 -6,0 L 0 -175 L 6 0 C 4,3 1,5 0,5"
+          d={`M 0 5 C -1,5 -4,3 -6,0 L 0 -${height} L 6 0 C 4,3 1,5 0,5`}
           stroke="black"
```

Comment:
`d` path attribute is not an svg style attribute, sorry for the confusion. The calculation of the needle path should happen inside the needle component and be based on some props passed in by VictoryGauge (i.e. the radius of the gauge so you make a reasonably sized needle for your gauge). This way users can define a custom needle component and calculate their own path